lib/active_model/serializer/adapter/json.rb

```diff
@@ -7,37 +7,13 @@ class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
```

Comment:
@NullVoxPopuli I agree on that, neither do I. But my question was why not get rid of the instance variable altogether and simply use a local variable?